pandas/_libs/testing.pyx

```diff
@@ -21,7 +21,11 @@ from pandas.core.dtypes.missing import (
 
 
 cdef bint isiterable(obj):
-    return hasattr(obj, '__iter__')
+    if hasattr(obj, '__iter__'):
```

Comment:
I'm not completely convinced this is an appropriate fix to simply exclude 0-D arrays here. This might have unintended consequences where we want to treat 0-D arrays as iterable.

Reply:
I've changed it to use try/except. Is this an appropiate fix?